platforms/documentation/docs/src/docs/userguide/developing-plugins/tutorial/part7_use_consumer_project.adoc

```diff
@@ -0,0 +1,147 @@
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[part7_use_consumer_project]]
+= Part 7: Use a Consumer Project
+
+Learn how to set up a dummy consumer project to test your plugin without publishing it.
+
+****
+**In this section, you'll:**
+
+- Create a `consumer` project to use your plugin.
+- Configure a **Composite Build** to link the `consumer` and `plugin` projects.
+- Test your plugin in a real-world scenario.
+****
+
+[[part7_begin]]
+== Step 0: Before you Begin
+
+1. You initialized your plugin in <<part1_gradle_init_plugin.adoc#part1_begin,part 1>>.
+2. You added an extension to your plugin in <<part2_add_extension.adoc#part2_begin,part 2>>.
+3. You created a custom task in <<part3_create_custom_task.adoc#part3_begin, part3>>.
+4. You wrote a unit test in <<part4_unit_test.adoc#part4_begin,part 4>>.
+5. You added a dataflow action to your plugin in <<part5_add_dataflow_action.adoc#part5_begin,part 5>>.
+6. You wrote a functional test in <<part6_functional_test.adoc#part6_begin,part 6>>.
+
+== Step 1: Create a Consumer Project
+
+We will now create a separate `consumer` project.
+This is a best practice for testing plugins because it simulates how a real user would apply your plugin to their own project.
+It also allows you to test your plugin without publishing it to a repository like the Gradle Plugin Portal.
+
+First, update your project structure to include a new `consumer` directory:
+
+[.multi-language-sample]
+=====
+[source, kotlin]
+----
+.
+├── gradlew
+├── settings.gradle.kts
+├── plugin/
+└── consumer/   //<1>
+    ├── settings.gradle.kts
+    └── build.gradle.kts
+----
+<1> Create a new `consumer` directory with its own build files.
+=====
+[.multi-language-sample]
+=====
+[source, groovy]
+----
+.
+├── gradlew
+├── settings.gradle
+├── plugin/
+└── consumer/   //<1>
+    ├── settings.gradle
+    └── build.gradle
+----
+<1> Create a new `consumer` directory with its own build files.
+=====
+
+Next, we'll configure a **Composite Build** by updating the root `settings.gradle(.kts)` file.
+This tells Gradle to treat your `plugin` and `consumer` projects as part of a single, coordinated build.
+
+====
+include::sample[dir="snippets/plugins/plugin-tutorial/kotlin",files="settings.gradle.kts[]"]
+include::sample[dir="snippets/plugins/plugin-tutorial/groovy",files="settings.gradle[]"]
+====
+
+The key line here is `includeBuild("plugin")`.
```

Comment:
I would also mention that this is done inside pluginManagement

Reply:
Done.